src/ComServerHelpers/WinRtServer.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using ComServerHelpers.Internal;
+using ComServerHelpers.Internal.Windows;
+using Windows.Win32.Foundation;
+using Windows.Win32.System.Com;
+using Windows.Win32.System.WinRT;
+using static Windows.Win32.PInvoke;
+//using unsafe DllActivationCallback = delegate* unmanaged[Stdcall]<Windows.Win32.System.WinRT.HSTRING, Windows.Win32.System.WinRT.IActivationFactory**, Windows.Win32.Foundation.HRESULT>;
+
+namespace ComServerHelpers;
+
+/// <summary>
+/// An Out of Process Windows Runtime Server.
+/// </summary>
+/// <remarks>
+/// <para>Allows for types to be created using WinRT activation instead of COM activation like <see cref="ComServer"/>.</para>
+/// <para>Typical usage is to call from an <see langword="await"/> <see langword="using"/> block, using <see cref="WaitForFirstObjectAsync"/> to not close until it is safe to do so.</para>
+/// <code language="cs">
+/// <![CDATA[
+/// using (WinRtServer server = new WinRtServer())
+/// {
+///     server.RegisterClass<RemoteThing>();
+///     server.Start();
+///     await server.WaitForFirstObjectAsync();
+/// }
+/// ]]>
+/// </code>
+/// </remarks>
+/// <see cref="IDisposable"/>
+/// <threadsafety static="true" instance="false"/>
+[SupportedOSPlatform("windows8.0")]
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1724", Justification = "No better idea")]
+public sealed class WinRtServer : IDisposable
+{
+    /// <summary>
+    /// Mapping of Activatable Class IDs to activation factories and their <see cref="ComWrappers"/> implementation.
+    /// </summary>
+    private readonly Dictionary<string, (BaseActivationFactory Factory, ComWrappers Wrapper)> factories = [];
+
+    private readonly unsafe DllGetActivationFactory activationFactoryCallbackWrapper;
+
+    private unsafe readonly delegate* unmanaged[Stdcall]<HSTRING, IActivationFactory**, HRESULT> activationFactoryCallbackPointer;
+    private readonly StrategyBasedComWrappers comWrappers = new();
```

Comment:
I see many instantiations of this type. Are they all needed? Can they be collapsed to a single for this library?